odoo/sql_db.py

```diff
@@ -91,6 +91,8 @@ def __init__(self, cr):
         self._cr = cr
         self.closed = False
         cr.execute('SAVEPOINT "%s"' % self.name)
+        if hasattr(self._cr, '_increase_savepoint_count'):
```

Comment:
In _check_savepoint, the savepoint is directly created with psycopg cursor object
`self._savepoint = Savepoint(self._cursor._obj)` and thus not have the counter. We don't really need to track those savepoint in the test case